terraform/environment/region/modules/schedule_runner/lambda.tf

```diff
@@ -0,0 +1,197 @@
+module "schedule_runner" {
+  source      = "../lambda"
+  lambda_name = "schedule-runner"
+  description = "Function to run scheduled tasks on a schedule set by EventBridge Scheduler"
+  environment_variables = {
+    EVENT_BUS_NAME             = var.event_bus_name
+    GOVUK_NOTIFY_BASE_URL      = "https://api.notifications.service.gov.uk"
+    GOVUK_NOTIFY_IS_PRODUCTION = data.aws_default_tags.current.tags.environment-name == "production" ? "1" : "0"
+    LPAS_TABLE                 = var.lpas_table.name
+    SEARCH_ENDPOINT            = var.search_endpoint
+    SEARCH_INDEX_NAME          = var.search_index_name
+    SEARCH_INDEXING_DISABLED   = 1
+  }
+  image_uri            = "${var.lambda_function_image_ecr_url}:${var.lambda_function_image_tag}"
+  aws_iam_role         = var.schedule_runner_lambda_role
+  environment          = data.aws_default_tags.current.tags.environment-name
+  kms_key              = data.aws_kms_alias.cloudwatch_application_logs_encryption.target_key_arn
+  iam_policy_documents = [data.aws_iam_policy_document.api_access_policy.json]
+  timeout              = 360
+  memory               = 1024
+  vpc_config = {
+    subnet_ids         = var.vpc_config.subnet_ids
+    security_group_ids = var.vpc_config.security_group_ids
+  }
+  providers = {
+    aws.region = aws.region
+  }
+}
+
+data "aws_iam_policy_document" "api_access_policy" {
+  statement {
+    sid       = "allowApiAccess"
+    effect    = "Allow"
+    resources = var.allowed_api_arns
+    actions = [
+      "execute-api:Invoke",
+    ]
+  }
+}
+
+resource "aws_scheduler_schedule" "schedule_runner_hourly" {
```

Comment:
I think you're missing a region provider here, it'll pass and then fail to apply because it doesn't have permission to create that resource in the identity account (your user that is used when no provider is given)